modules/layers/src/text-layer/text-layer.js

```diff
@@ -108,7 +108,16 @@ export default class TextLayer extends CompositeLayer {
 
   // eslint-disable-next-line complexity
   updateState({props, oldProps, changeFlags}) {
-    const fontChanged = this._fontChanged(oldProps, props);
+    const textChanged =
+      changeFlags.dataChanged ||
+      (changeFlags.updateTriggersChanged &&
+        (changeFlags.updateTriggersChanged.all || changeFlags.updateTriggersChanged.getText));
+
+    if (textChanged) {
+      this._updateText();
+    }
+
+    const fontChanged = textChanged || this._fontChanged(oldProps, props);
```

Comment:
Is it not possible for the text to change in such a way so that the font is unchanged? I.e. the text grows but no new characters are used. In this case the font atlas will be updated unnecessarily.

Reply:
The FontAtlasManager checks for the difference between the existing font map and the incoming `characterSet`. Only new characters are added.